adc/consumer.py

```diff
@@ -154,6 +154,14 @@ def _stream_until_eof(self,
             try:
                 for m in messages:
                     err = m.error()
+                    # A new message may arrive from a previously removed topic/partition,
+                    # in which case it must be re-added
+                    if m.topic() not in active_partitions:
+                        active_partitions[m.topic()] = set()
+                    partition_set = active_partitions[m.topic()]
+                    if m.partition() not in partition_set:
+                        partition_set.add(m.partition())
```

Comment:
I'd just add it to the set w/o the explicit `if` check above -- makes the code shorter & I suspect most of the cost is in the lookup of whether the element exists in the set anyway (which must be done for both the if statement and the .add).

Reply:
For some reason I was apparently convinced that would be somehow unsafe. Redundant check is now removed.